### PR TITLE
Update Dependencies

### PR DESCRIPTION
**Pull request by cdr-chakotay**
Only a few dependency updates need further discussion. There are to name:

-  the TUS Client
-  io.socket:socket.io-client => There is a need to clarify, which Version is running on the transloadit server

Some dependencies could not be updated to their newest versions. I have chosen the lastest compatible ones:
- I decided to stay at jUnit4 for the moment to avoid remodeling all the tests. The latest version is from January so it seems not to be abandoned.
- The Mockserver has a special version to support jUnit 4 in the future,I switched to it.

All the other I have updated to the most recent version available.

closing #46

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -23,16 +23,20 @@ allprojects {
...
     implementation 'io.tus.java.client:tus-java-client:0.4.1'
-    implementation 'joda-time:joda-time:2.9.7'
-    implementation 'com.squareup.okhttp3:okhttp:3.6.0'
+    implementation 'joda-time:joda-time:2.10.10'
+    implementation 'com.squareup.okhttp3:okhttp:4.9.1'
     implementation 'org.json:json:20210307'
-    implementation 'commons-codec:commons-codec:1.9'
+    implementation 'commons-codec:commons-codec:1.15'
+    // implementation 'io.socket:socket.io-client:2.0.1' Check Version
     implementation 'io.socket:socket.io-client:1.0.0'
```
Comment (cdr-chakotay) [inline build.gradle:33]: We have to check with which version(s) of socket.io we have to be compatible with
[https://socketio.github.io/socket.io-client-java/installation.html](https://socketio.github.io/socket.io-client-java/installation.html)

Comment (cdr-chakotay) [reply build.gradle:33]: Will go for 1.0.1 as we are on SocketIO 2.x on the Server side

**Comment by Acconut** [comment]
Wow, thanks for taking a stab at this! Is this ready to be merged? It is currently still marked as a draft

**Comment by cdr-chakotay** [comment]
> Wow, thanks for taking a stab at this! Is this ready to be merged? It is currently still marked as a draft

The TUS dependency still remains open, wanted to have a talk to you if upgrading to a minor release makes sense here :) 
Besides that, I haven't discovered any bigger bugs so far. 
So we shall be able to merge this fairly soon. 

We could also think about a new release maybe => but we will have minor compatibility issues due to the updated AssemblyListener Interface, which needs to be implemented by the customers.

**Pull request merged**